### DISCUSSION TITLE
How to save sleuth.db in an alternative path?

### DISCUSSION THREAD

**Discussion by claytoncova** [sepinf-inc/IPED#337] [Q&A] [answered]
Hello there!

Due to larger file size and amount of outputs in a nowadays extractions, we decide to use a Storage Server as output to IPED operation, trough NFS on ethernet LAN.

There´sa problem with this approach, IPED stop in an error, that boilers down to sleuth.db file lock issue in a remote server.

Is it possible to change or define a distinct path to save sleuth.db, in a meantime the extraction is performed?

Ps .: At this moment, we´re using a non neat approach to work around the issue, using symbolic links and remount points.

**Comment by lfcnassif** [comment, marked as answer]
Currently it is not possible. I already thought about storing it temporarily in local temp folder while processing, to workaround network file system lock issues, a lot of them are very broken. There are some details to make --append and --continue options to work with that approach...

Actually, I don't know exactly why these kind of lock issues happen when the output is on a network share, because the sleuth.db sqlite database is accessed from multiple processes/threads from the **same** machine, so the OS should handle this and make it transparent. Note those issues does not happen with Lucene index, that uses mmap, which is more complex, but SQLite site IMHO explicitly says WAL log (that uses mmap) does not work on network FS, so we use default rollback journal. Maybe it is a SQLite issue...

I suggest you trying to update your NFS implementation to NFS v4, if it is <=v3, a lot of lock issues were fixed, and I think a colleague of mine uses it with no such problems.

Regards
